### PR TITLE
[Enhancement] Add versioning for PersistentIndex file

### LINKED ISSUE #4768: Add versioning for PersistentIndex file to support future upgrade compatibility
Add versioning for PersistentIndex file to support future upgrade compatibility, add an index file version field in meta

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] others

## Which issues of this PR fixes ：

Fixes #4768

## Problem Summary(Required) ：

Add versioning for PersistentIndex file to support future upgrade compatibility, add an index file version field in meta

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/storage/persistent_index.h [hunk 1]
```diff
@@ -21,6 +21,13 @@ class Schema;
 class Column;
 } // namespace vectorized
 
+// Add version for persistent index file to support future upgrade compatibility
+// There is only one version for now
+enum IndexFileVersion {
+    UNKNOWN = 0,
+    VERSION_1,
```
Comment (imay) [inline be/src/storage/persistent_index.h:28]:
```suggestion
    PERSISTENT_INDEX_VERSION_1,
```

version_1 is too common

File: be/src/storage/persistent_index.h [hunk 2]
```diff
@@ -21,6 +21,13 @@ class Schema;
 class Column;
 } // namespace vectorized
 
+// Add version for persistent index file to support future upgrade compatibility
+// There is only one version for now
+enum IndexFileVersion {
```
Comment (imay) [inline be/src/storage/persistent_index.h:26]: we have bitmap index too. should make this clearer

File: be/src/storage/persistent_index.h (hunk 1 above)
Comment (sevev) [reply be/src/storage/persistent_index.h]: > version_1 is too common

done, changed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: gensrc/proto/persistent_index.proto [hunk 3]
```diff
@@ -18,6 +18,7 @@ message IndexWalMetaPB {
 message MutableIndexMetaPB {
     IndexSnapshotMetaPB snapshot = 2;
     repeated IndexWalMetaPB wals = 3;
+    uint32 format_version = 4;
```
Comment (decster) [inline gensrc/proto/persistent_index.proto:21]: this feature is not released, so better to arrange format to first, uint32 format_version = 1;

Comment (sevev) [reply gensrc/proto/persistent_index.proto]: > this feature is not released, so better to arrange format to first, uint32 format_version = 1;

done, change in https://github.com/StarRocks/starrocks/pull/5544/commits/a90fdab638c491c937a520442e2ab2c84079f8c9

**Review by decster — APPROVED** [review]

**Comment by sevev** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by chaoyli — APPROVED** [review]

**Pull request merged**